makefiles/deprecated_boards.inc.mk

```diff
@@ -0,0 +1,11 @@
+# Add deprecated modules here
+# Keep this list ALPHABETICALLY SORTED!!!!
+ifeq ($(MAKELEVEL),0)
+  DEPRECATED_BOARDS += 6lowpan-clicker
+  DEPRECATED_BOARDS += pic32-wifire
+
+  ifneq (,$(filter $(DEPRECATED_BOARDS),$(BOARD)))
```

Comment:
We are deprecating the CPU, to make it easy to implement I just showed it on all the boards using the cpu. Keep in mind this is a helper function when using the board.  The proper deprecation is in the doxygen docs which states both the boards and cpu are being deprecated.

This will be gone in 2 release cycles anyways.

Reply:
> The proper deprecation is in the doxygen docs which states both the boards and cpu are being deprecated.

I get this. I find it strange the we put this deprecation warning system in place but fail to warn CPU-only users who may not be re-reading the docs. Moreover, by triggering this message when mips CPU is used it will also warn the board users.

Reply:
Would this patch be OK then? I do agree with your concern, and deprecating CPU/BOARDs will happen again so it makes sense to have something in place for both.

```diff
diff --git a/Makefile.include b/Makefile.include
index 0f9a5199d4..68be673d70 100644
--- a/Makefile.include
+++ b/Makefile.include
@@ -389,14 +389,15 @@ endif
 # Add standard include directories
 INCLUDES += -I$(RIOTBASE)/core/include -I$(RIOTBASE)/drivers/include -I$(RIOTBASE)/sys/include

+# process provided features
+include $(RIOTBASE)/Makefile.features
+
 ifneq ($(RIOT_CI_BUILD),1)
   # Warn about used deprecated boards
   include $(RIOTMAKE)/deprecated_boards.inc.mk
+  include $(RIOTMAKE)/deprecated_cpus.inc.mk
 endif

-# process provided features
-include $(RIOTBASE)/Makefile.features
-
 # mandatory includes!
 include $(RIOTMAKE)/pseudomodules.inc.mk
 include $(RIOTMAKE)/defaultmodules.inc.mk
diff --git a/makefiles/deprecated_cpus.inc.mk b/makefiles/deprecated_cpus.inc.mk
new file mode 100644
index 0000000000..6198781b09
--- /dev/null
+++ b/makefiles/deprecated_cpus.inc.mk
@@ -0,0 +1,9 @@
+# Add deprecated modules here
+# Keep this list ALPHABETICALLY SORTED!!!!
+ifeq ($(MAKELEVEL),0)
+  DEPRECATED_CPUS += mips_pic32mx
+  ifneq (,$(filter $(DEPRECATED_CPUS),$(CPU)))
+    $(shell $(COLOR_ECHO) "$(COLOR_RED)Deprecated cpu: $(COLOR_RESET)"\
+                          "$(CPU)" 1>&2)
+  endif
+endif
```

Reply:
Thanks, yes it looks good to me. We should add `mips_pic32mx` as well.

Reply:
Done

Reply:
Thanks